lib/game/widget_builder.dart

```diff
@@ -0,0 +1,202 @@
+          : null,
+
+      // Force presses
+      onForcePressStart: game is ForcePressDetector
+          ? (ForcePressDetails d) =>
+              (game as ForcePressDetector).onForcePressStart(d)
+          : null,
+      onForcePressPeak: game is ForcePressDetector
+          ? (ForcePressDetails d) =>
+              (game as ForcePressDetector).onForcePressPeak(d)
+          : null,
+      onForcePressUpdate: game is ForcePressDetector
+          ? (ForcePressDetails d) =>
+              (game as ForcePressDetector).onForcePressUpdate(d)
+          : null,
+      onForcePressEnd: game is ForcePressDetector
+          ? (ForcePressDetails d) =>
+              (game as ForcePressDetector).onForcePressEnd(d)
+          : null,
+
+      // Pan
+      onPanDown: game is PanDetector
+          ? (DragDownDetails d) => (game as PanDetector).onPanDown(d)
+          : null,
+      onPanStart: game is PanDetector
+          ? (DragStartDetails d) => (game as PanDetector).onPanStart(d)
+          : null,
+      onPanUpdate: game is PanDetector
+          ? (DragUpdateDetails d) => (game as PanDetector).onPanUpdate(d)
+          : null,
+      onPanEnd: game is PanDetector
+          ? (DragEndDetails d) => (game as PanDetector).onPanEnd(d)
+          : null,
+      onPanCancel: game is PanDetector
+          ? () => (game as PanDetector).onPanCancel()
+          : null,
+
+      // Scales
+      onScaleStart: game is ScaleDetector
+          ? (ScaleStartDetails d) => (game as ScaleDetector).onScaleStart(d)
+          : null,
+      onScaleUpdate: game is ScaleDetector
+          ? (ScaleUpdateDetails d) => (game as ScaleDetector).onScaleUpdate(d)
+          : null,
+      onScaleEnd: game is ScaleDetector
+          ? (ScaleEndDetails d) => (game as ScaleDetector).onScaleEnd(d)
+          : null,
+
+      child: Container(
+          color: game.backgroundColor(),
+          child: Directionality(
+              textDirection: TextDirection.ltr,
+              child: EmbeddedGameWidget(game))),
+    );
+  }
+}
+
+class OverlayGameWidget extends StatefulWidget {
+  final Widget child;
+  final HasWidgetsOverlay game;
+
+  OverlayGameWidget({this.child, this.game});
+
+  @override
+  State<StatefulWidget> createState() => _OverlayGameWidgetState(game);
+}
+
+class _OverlayGameWidgetState extends State<OverlayGameWidget> {
+  final Map<String, Widget> _overlays = {};
+
+  _OverlayGameWidgetState(HasWidgetsOverlay game) {
+    game.widgetOverlayController.stream.listen((overlay) {
+      setState(() {
+        if (overlay.widget == null) {
+          _overlays.remove(overlay.name);
+        } else {
+          _overlays[overlay.name] = overlay.widget;
+        }
+      });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Directionality(
+        textDirection: TextDirection.ltr,
+        child: Stack(
+            children: [
+          widget.child,
+        ]..addAll(_overlays.values.toList())));
```

Comment:
you can use the `...` operator for this (`[widget.child, ..._overlays.values[.toList()]])`)

Reply:
To use that operator, we need to increase the min required dart version from 2.2.0 to 2.3.0, what do you think about this? I don't mind really, but that is probably kind of a minor breaking change.

Reply:
I'm ok with it, we can bump a minor in the next version :)

Reply:
okay, done then!